### ISSUE TITLE
"Most recent observation" aggregation mode

### ISSUE THREAD

**Issue by gmichalec-pandora** [prometheus/client_js#404]
Is it possible to use clustering with a gauge-style metric? 
From my testing, it seems like you can't use a gauge-style metric - i.e. a metric value that could either increase or decrease - and get results reported in a predictable manner. 
I've attempted to use both the 'first' aggregator, as well as a custom 'last' aggregator. In either case, it seems that the metric value returned was a random selection from the pool of workers (most likely in the order they answered the request event). **For a gauge-style metric to work properly, I would need to get the most-recent value reported, however collection time does not appear to be present in the metrics list returned to the aggregator.**
 
Based on that research, I'm assuming there is no way to implement a 'most-recent' aggregator, but thought I would ask to confirm.

Thanks for writing a great library! Works perfectly in every other respect.

**Comment by zbjornson** [comment]
> I would need to get the most-recent value reported

Yeah, there's no way to do that with the current aggregator code unfortunately. 😞

Renaming the issue and marking it as an enhancement in a moment.

**Comment by jdmarshall** [comment]
That's not really how telemetry tools generally work. 

That said, there's some tricks one can do with Exemplars to get a little bit more metadata with a sample. It's kind of at cross purposes for what you're asking for, but you can at least figure out which sample you're actually looking at.
